### PR TITLE
Feature/role assign idam role merge

### PR DESCRIPTION
**Pull request by udaydenduluri33**
https://tools.hmcts.net/jira/browse/EUI-3643
https://tools.hmcts.net/jira/browse/EUI-4497

Merge IDAM roles with Role management API roles on User authentication and add it to user object sent to Angular

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/app/containers/app-header/app-header.component.spec.ts [hunk 1]
```diff
@@ -5,11 +5,12 @@ import { FeatureToggleService } from '@hmcts/rpx-xui-common-lib';
 import { Action, Store, StoreModule } from '@ngrx/store';
 import { BehaviorSubject, combineLatest, Observable, of, Subscription } from 'rxjs';
 import { AppConstants } from 'src/app/app.constants';
+import { Theme } from 'src/app/models/theme.model';
```
Comment (connorpgpmcelroy) [inline src/app/containers/app-header/app-header.component.spec.ts:8]: src in new line and the one above

File: src/app/containers/app-header/app-header.component.spec.ts [hunk 2]
```diff
@@ -183,32 +184,38 @@ describe('AppHeaderComponent', () => {
         ];
 
         const defaultTheme = AppConstants.DEFAULT_USER_THEME;
-
-        expect(component.getUsersTheme(userRoles, themes, defaultTheme)).toEqual(themes[0]);
+        const userTypeRoles = { Solicitor: [] }
```
Comment (connorpgpmcelroy) [inline src/app/containers/app-header/app-header.component.spec.ts:187]: think this should end in semi-colon

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Small changes required - Apart from that, looks good to me

File: src/app/containers/app-header/app-header.component.spec.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/app/containers/app-header/app-header.component.spec.ts:187]: Done

File: src/app/containers/app-header/app-header.component.spec.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/app/containers/app-header/app-header.component.spec.ts]: Done

**Review by connorpgpmcelroy — APPROVED** [review]
Comments addressed - Looks good to me

**Review by kuda1992 — APPROVED** [review]

**Pull request merged**